sdk/aqueduct/llm_wrapper.py

```diff
@@ -0,0 +1,220 @@
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import pandas as pd
+from aqueduct.artifacts.base_artifact import BaseArtifact
+from aqueduct.constants.enums import RuntimeType
+from aqueduct.decorator import op
+from aqueduct.error import InvalidUserArgumentException
+from aqueduct.integrations.dynamic_k8s_integration import DynamicK8sIntegration
+from aqueduct.utils.utils import generate_engine_config
+
+from aqueduct import globals
+
+supported_llms = [
+    "llama_7b",
+    "vicuna_7b",
+    "dolly_v2_3b",
+    "dolly_v2_7b",
+]
+
+resource_requests = {
+    "llama_7b": {
+        "memory": "16GB",
+        "gpu_resource_name": "nvidia.com/gpu",
+    },
+    "vicuna_7b": {
+        "memory": "32GB",
+        "gpu_resource_name": "nvidia.com/gpu",
+    },
+    "dolly_v2_3b": {
+        "memory": "8GB",
+        "gpu_resource_name": "nvidia.com/gpu",
+    },
+    "dolly_v2_7b": {
+        "memory": "16GB",
+        "gpu_resource_name": "nvidia.com/gpu",
+    },
+}
+
+
+def _generate_llm_op(
+    llm_name: str, column_name: Optional[str] = None, output_column_name: Optional[str] = None
+) -> Callable[
+    [Union[str, List[str], pd.DataFrame], Dict[str, Any]], Union[str, List[str], pd.DataFrame]
+]:
+    def use_llm(
+        messages: Union[str, List[str]], parameters: Dict[str, Any] = {}
+    ) -> Union[str, List[str]]:
+        if not (isinstance(messages, str) or isinstance(messages, list)):
+            raise ValueError("The input must be a string or list of strings.")
+
+        module = __import__("aqueduct_llm", fromlist=[llm_name])
+        llm = getattr(module, llm_name)
+
+        if "prompt" in parameters:
+            prompt = parameters["prompt"]
+            if not isinstance(prompt, str):
+                raise ValueError("The 'prompt' parameter must be a string.")
+
+            if "{text}" not in prompt:
```

Comment:
Can you explain the significance of `"{text}"` is this a keyword in LLM prompts?

Reply:
"If the prompt contains {text}, we will replace {text} with the input string(s) before sending to the LLM. If the prompt does not contain {text}, we will prepend the prompt to the input string(s) before sending to the LLM."